### ISSUE TITLE
inline bash comment breaks version detection

### ISSUE THREAD

**Issue by lkwg82** [hadolint/hadolint#466]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

no issues

### Actual behavior

Dockerfile:5 DL3018 Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

### Steps to reproduce the behavior

```bash
$ hadolint -v
Haskell Dockerfile Linter 1.18.0-no-git
```

**Dockerfile** (if relevant)

```Dockerfile
FROM alpine

RUN apk --no-cache add 
          jq=1.6-r1 \
          git=2.26.2-r0 `# needed for glab` \
          py-pip=20.1.1-r0 \

```

**Comment by m-ildefons** [comment]
I'd argue that this is actually correct behavior. The code within the backticks is actually executed in shell and the output of that code is interpreted as argument to `apk`, in your case.
There is no simple way of determining whether or not the code within the backticks produces output and what that output is without executing it (hello decidability?).

In your case, the code within the backticks is just a comment and produces no output. Regardless, the output of the backticked code would be taken as a package to be installed an has not version attached to it.

Consider:
```Dockerfile
RUN apk --no-cache add jq=1.6-r1 `echo -n ""`
```

This has the same behaviour as:
```Dockerfile
RUN apk --no-cache add jq=1.6-r1 `# comment`
```

IMO you could argue, that using backticks to move a comment from the Dockerfile to within the shell that executes commands during building of the image is bad practice. And that the message is misleading and confusing, but it is not wrong.
Try this instead:

```Dockerfile
FROM alpine
RUN apk --no-cache add \
          jq=1.6-r1 \
# needed for glab:
          git=2.26.2-r0 \
          py-pip=20.1.1-r0 \
```

**Comment by lkwg82** [comment]
@m-ildefons really like ur comment on the next line

So this is regarding readability a better outcome. Just need to relearn my habit.

Thx anyway
